pkg/framework/plugins/removepodsviolatinginterpodantiaffinity/pod_antiaffinity.go

```diff
@@ -14,80 +14,88 @@ See the License for the specific language governing permissions and
+	"sigs.k8s.io/descheduler/pkg/framework"
 	"sigs.k8s.io/descheduler/pkg/utils"
 
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
 )
 
-func validateRemovePodsViolatingInterPodAntiAffinityParams(params *api.StrategyParameters) error {
-	if params == nil {
-		return nil
-	}
-
-	// At most one of include/exclude can be set
-	if params.Namespaces != nil && len(params.Namespaces.Include) > 0 && len(params.Namespaces.Exclude) > 0 {
-		return fmt.Errorf("only one of Include/Exclude namespaces can be set")
-	}
-	if params.ThresholdPriority != nil && params.ThresholdPriorityClassName != "" {
-		return fmt.Errorf("only one of thresholdPriority and thresholdPriorityClassName can be set")
-	}
+const PluginName = "RemovePodsViolatingInterPodAntiAffinity"
 
-	return nil
+// RemovePodsViolatingInterPodAntiAffinity evicts pods on the node which violate inter pod anti affinity
+type RemovePodsViolatingInterPodAntiAffinity struct {
+	handle    framework.Handle
+	args      *componentconfig.RemovePodsViolatingInterPodAntiAffinityArgs
+	podFilter podutil.FilterFunc
 }
 
-// RemovePodsViolatingInterPodAntiAffinity evicts pods on the node which are having a pod affinity rules.
-func RemovePodsViolatingInterPodAntiAffinity(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, evictorFilter *evictions.EvictorFilter, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc) {
-	if err := validateRemovePodsViolatingInterPodAntiAffinityParams(strategy.Params); err != nil {
-		klog.ErrorS(err, "Invalid RemovePodsViolatingInterPodAntiAffinity parameters")
-		return
+var _ framework.Plugin = &RemovePodsViolatingInterPodAntiAffinity{}
+var _ framework.DeschedulePlugin = &RemovePodsViolatingInterPodAntiAffinity{}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	interPodAntiAffinityArgs, ok := args.(*componentconfig.RemovePodsViolatingInterPodAntiAffinityArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type RemovePodsViolatingInterPodAntiAffinityArgs, got %T", args)
 	}
 
 	var includedNamespaces, excludedNamespaces sets.String
-	var labelSelector *metav1.LabelSelector
-	if strategy.Params != nil {
-		if strategy.Params.Namespaces != nil {
-			includedNamespaces = sets.NewString(strategy.Params.Namespaces.Include...)
-			excludedNamespaces = sets.NewString(strategy.Params.Namespaces.Exclude...)
-		}
-		labelSelector = strategy.Params.LabelSelector
+	if interPodAntiAffinityArgs.Namespaces != nil {
+		includedNamespaces = sets.NewString(interPodAntiAffinityArgs.Namespaces.Include...)
+		excludedNamespaces = sets.NewString(interPodAntiAffinityArgs.Namespaces.Exclude...)
 	}
 
 	podFilter, err := podutil.NewOptions().
+		WithFilter(handle.Evictor().Filter).
```

Comment:
`handle.Evictor().Filter` filter needs to go out since the plugin needs to take into account also pods which are not evictable to get the overall anti-affinity view among all pods.

Reply:
Good catch! I totally missed this one